internal/locate/region_request.go

```diff
@@ -1931,6 +1936,9 @@ func (s *RegionRequestSender) onRegionError(
 	)
 
 	if s.replicaSelector != nil {
+		if errLabel == "mismatch_peer_id" {
+			s.replicaSelector.invalidateRegion()
```

Comment:
In this case it seems we should return `false, nil` to let the region reloading instead of retry directly.